### PR TITLE
remove OutBuffer.peekSlice(), replace with []

### PR DESCRIPTION
**Pull request by WalterBright**
It's redundant with the `opSlice()` function.

### CODE REVIEW

File: src/dmd/dmangle.d [hunk 1]
```diff
@@ -601,7 +601,7 @@ public:
         mangleDecl(d);
         debug
         {
-            const slice = buf.peekSlice();
+            const slice = (*buf)[];
```
Comment (jacob-carlborg) [inline src/dmd/dmangle.d]: If you explicitly call `opSlice` you don’t need the parentheses or the star.

File: src/dmd/mars.d [hunk 2]
```diff
@@ -635,7 +635,7 @@ private int tryMain(size_t argc, const(char)** argv, ref Param params)
         foreach (i; 1 .. modules[0].aimports.dim)
             semantic3OnDependencies(modules[0].aimports[i]);
 
-        const data = ob.peekSlice();
+        const data = (*ob)[];
```
Comment (jacob-carlborg) [inline src/dmd/mars.d]: If you explicitly call `opSlice` you don’t need the parentheses or the star.

Comment (WalterBright) [reply src/dmd/mars.d]: That's why ob should probably be made a ref instead of a pointer. But this is an issue for the caller, not OutBuffer.

Comment (rainers) [reply src/dmd/mars.d]: `ref` can only be used on parameters, not on locals or fields.

File: src/dmd/mars.d [hunk 3]
```diff
@@ -1338,7 +1338,7 @@ extern(C) void printGlobalConfigs(FILE* stream)
                 buf.printf("%s ", flag);
         }
 
-        auto res = buf.peekSlice() ? buf.peekSlice()[0 .. $ - 1] : "(none)";
+        auto res = buf[] ? buf[][0 .. $ - 1] : "(none)";
```
Comment (rainers) [inline src/dmd/mars.d]: Nits: `buf[]` is not the best way to check for length 0, It returns true after writing to the buffer and resetting it with `reset()`.
`buf[][0 .. $ - 1]` could be written as `buf[0 .. $ - 1]` if `opDollar` is also defined in OutBuffer.

**Review by rainers — APPROVED** [review]

**Pull request merged**